cli/src/main/java/com/devonfw/tools/ide/process/ProcessContextImpl.java

```diff
@@ -167,14 +167,15 @@ public ProcessResult run(ProcessMode processMode) {
 
       this.processBuilder.command(args);
 
+      List<LogEvent> logs = new ArrayList<>();
       List<String> out = null;
       List<String> err = null;
 
       Process process = this.processBuilder.start();
 
       if (processMode == ProcessMode.DEFAULT_CAPTURE) {
-        CompletableFuture<List<String>> outFut = readInputStream(process.getInputStream());
-        CompletableFuture<List<String>> errFut = readInputStream(process.getErrorStream());
+        CompletableFuture<List<String>> outFut = readInputStream(process.getInputStream(), false, logs);
+        CompletableFuture<List<String>> errFut = readInputStream(process.getErrorStream(), true, logs);
         out = outFut.get();
         err = errFut.get();
```

Comment:
Yes, that is the plan but we should keep the existing API methods `getOut()` and `getErr()` to avoid that all using code also needs to get refactored. But you should take a `List<OutputMessage> output` instead and make that available via a new getter. The old getters can be implemented with streams and lambdas.